chapter-16/a-monster/a-monster.js

```diff
@@ -0,0 +1,64 @@
+import { DOMDisplay, Vec, levelChars, runLevel, Level, State } from "../lib/game.js";
+
+class Monster {
+    constructor(pos) {
+        this.pos = pos;
+    }
+
+    get type() { return "monster"; }
+
+    static create(pos) {
+      return new Monster(pos.plus(new Vec(0, -1)));
+    }
+
+    update(time, state) {
+        const player = state.player;
+      const playerPosX = player.pos.x;
+      let newPos;
+      let speed = 0.05;
```

Comment:
From the `Player` class we can tell that it has a constant speed of `7` that is multiplied with the provided `time` input on every `Player.update` call to get the amount of distance to move on the x/y axis.

The `time` value is usually a number between 0.016 and 0.018. To get approximately the same speed as you chose here, we would set the "speed constant" of the monster to `3`.

You can do this in the constructor of the `Monster` class by setting `this.speed=3`.

Then the new position vector of the monster is calculated like this:
```javascript
newMonsterPos = new Vec(monsterPos.x + (time * this.speed), monsterPos.y);
```

Reply:
Done!